install/operator/pkg/syndesis/configuration/configuration.go

```diff
@@ -498,3 +500,16 @@ func exposeVia3Scale(config map[string]string, syndesis *v1alpha1.Syndesis) {
 		}
 	}
 }
+
+func envDatavirtEnabled(syndesis *v1alpha1.Syndesis) *SyndesisEnvVarConfig {
+	if komodo := syndesis.Spec.Addons["komodo"]["enabled"]; komodo == "true" {
+		return &SyndesisEnvVarConfig{
+			Var:   EnvDatavirtEnabled,
+			Value: "1",
+		}
+	}
+	return &SyndesisEnvVarConfig{
```

Comment:
retunr `nil` here. The place to set defaults is: 
https://github.com/syndesisio/syndesis/blob/3e529cc10e9cf6ef83b21bbf7946ae222923d1f6/install/operator/pkg/syndesis/template/install.go#L84

Reply:
ok, a question, should I provide a default for the `syndesis.Spec.Addons["komodo"]["enabled"]` value?

The whole problem is with #6314 is that https://github.com/syndesisio/syndesis/blob/3e529cc10e9cf6ef83b21bbf7946ae222923d1f6/install/operator/pkg/generator/assets/infrastructure/03-syndesis-ui.yml.mustache#L122
references `Env.DATAVIRT_ENABLED` which, when unset, results in config.json like 
```
"datavirt": {
  "dvUrl": "/vdb-builder/v1/",
  "enabled": 
}
```
so we need to set the env var to either 0 or 1.

I'm thinking I should do it similarly to https://github.com/syndesisio/syndesis/blob/3e529cc10e9cf6ef83b21bbf7946ae222923d1f6/install/operator/pkg/syndesis/configuration/configuration.go#L483-L492
where we check the value from CR and set the env var accordingly.

wdyt?

Reply:
I think I got it now, will update shortly

Reply:
With latest change the operator correctly enables or disables datavirt when `komodo.enabled` is `true`/`false` or missing entirely.

Reply:
Yes you should set that 0 default value the same way as https://github.com/syndesisio/syndesis/blob/3e529cc10e9cf6ef83b21bbf7946ae222923d1f6/install/operator/pkg/syndesis/template/install.go#L107

only with your variable